src/Planners/include/BipedalLocomotion/Planners/UnicyclePlanner.h

```diff
@@ -0,0 +1,154 @@
+                          const Contacts::ContactList& right = {})
+        : left(left)
+        , right(right)
+    {
+    }
+
+    Contacts::ContactList left; ///< The list of left foot contacts;
+    Contacts::ContactList right; ///< The list of right foot contacts;
+};
+
+class BipedalLocomotion::Planners::UnicyclePlanner final
+    : public System::Advanceable<UnicyclePlannerInput, UnicyclePlannerOutput>
+{
+public:
+    UnicyclePlanner();
+
+    virtual ~UnicyclePlanner();
+
+    // clang-format off
+
+    /**
+     * Initialize the planner.
+     *
+     * @note The following parameters are required by the class:
+     *
+     * |          Name          |      Type      |   Default    | Mandatory |                    Description                    |
+     * | :--------------------: | :------------: | :----------: | :-------: | :-----------------------------------------------: |
+     * |    `sampling_time`     |     double     |      -       |    Yes    |         The sampling time of the planner          |
+     * |     `unicycleGain`     |     double     |     10.0     |    No     |     The main gain of the unicycle controller      |
+     * | `slowWhenTurningGain`  |     double     |     0.0      |    No     |    The turnin gain of the unicycle controller     |
+     * |  `referencePosition`   | list of double | (0.10, 0.00) |    No     | The reference position of the unicycle controller |
+     * |      `timeWeight`      |     double     |     1.0      |    No     |         The time weight of the OC problem         |
+     * |    `positionWeight`    |     double     |     1.0      |    No     |       The position weight of the OC problem       |
+     * |   `minStepDuration`    |     double     |      -       |    Yes    |          The minimum duration of a step           |
+     * |   `maxStepDuration`    |     double     |      -       |    Yes    |          The maximum duration of a step           |
+     * |   `nominalDuration`    |     double     |      -       |    Yes    |          The nominal duration of a step           |
+     * |    `minStepLength`     |     double     |      -       |    Yes    |           The minimum length of a step            |
+     * |    `maxStepLength`     |     double     |      -       |    Yes    |           The maximum length of a step            |
+     * |       `minWidth`       |     double     |      -       |    Yes    |             The minimum feet distance             |
+     * |     `nominalWidth`     |     double     |      -       |    Yes    |             The nominal feet distance             |
+     * |  `minAngleVariation`   |     double     |      -       |    Yes    |           The minimum unicycle rotation           |
+     * |  `maxAngleVariation`   |     double     |      -       |    Yes    |           The maximum unicycle rotation           |
+     * | `switchOverSwingRatio` |     double     |      -       |    Yes    | The ratio between step duration and stance phase  |
```

Comment:
Note, this is the ratio between the double and the single support phase. For example, 1.0 means that the single and double support phases are equal, while right now it would mean that the swing phase is null. The computation to get the percentage of the double support phase only is in https://github.com/robotology/unicycle-footstep-planner/blob/master/src/UnicycleGenerator.cpp#L564
Back in time, I preferred asking the user to set the ratio, so the swing time can never be zero.

Reply:
Oh thanks for the explanation. This parameter is the only one whose logic has been re-implemented in this PR, particularly in https://github.com/diegoferigo/bipedal-locomotion-framework/blob/7f65bc03289db32b0545c00f704310fd6fea607a/src/Planners/src/UnicyclePlanner.cpp#L463-L465.

The existing description _The ratio between step duration and stance phase_ is what is currently implemented, but as you noted it is different than the original meaning. We can either 1) keep the current logic and change the parameter name to avoid confusion, or 2) align with the previous logic.

Since we kept as much as possible backward-compatible configuration options, I'm going to proceed with 2).

Reply:
Done in 79339a5